File: MyProject/MinHeap.java
public class MinHeap {
    
    private final FilaDinamica fila;

    public MinHeap() {
        this.fila = new FilaDinamica();
    }

    public int tamanho() {
        return fila.getTamanho();
    }

    public void inserir(NoArvore noArvore) {
        fila.inserir(noArvore);
    }

    public NoArvore removerMinimo() {
        return fila.removerMinimo();
    }

    public void imprimirMinHeap() {
        fila.imprimirFila();
    }
}

File: MyProject/C++/MinHeap.cpp
#include "MinHeap.hpp"

MinHeap::MinHeap() {}

int MinHeap::tamanho() const { return fila.getTamanho(); }

void MinHeap::inserir(NoArvore* noArvore) { fila.inserir(noArvore); }

NoArvore* MinHeap::removerMinimo() { return fila.removerMinimo(); }

void MinHeap::imprimirMinHeap() const { fila.imprimirFila(); }